### ISSUE TITLE
Issue 2536 regarrding nested test classes not fully solved in 7.5

### ISSUE THREAD

**Issue by baubakg** [cbeust/testng#2722]
### TestNG Version
7.5

### Expected behavior
As the release notes state, issue #2536 is corrected by 7.5. However it is incorrect. One of the provided tests still fails.

#### Case
I am using annotation transformer to externalize test group definitions. I define a map of test + group at run time to my tests, and they are executed. This should work for all test definitions:

- Class level
- method level
- Nested Classes

### Actual behavior
Although 7.5 solved one issue no working in #2536 , there is still one issue that is not not working:

If we select a package in our tests and pick a nested class to be selected by the annotation transformer, the annotation transformer is completely ignored.

### Is the issue reproducible on runner?

- [x ] Maven
- [x ] Eclipse

### Test case sample
Please find attached my project (dedicated to this problem):
https://github.com/baubakg/testNGNestedClassIssue.git

The test class reproducing my issues is :
my.tests.GroupSelectorTests

Use cases showing the issue:

- my.tests.GroupSelectorTests.testNormalTestClass_packageIncludesNestedTestClass() (Fixed in 7.5)
- my.tests.GroupSelectorTests.testNestedTestClass_packageIncludesNormalTestClasses() (**FAILS in 7.5**)

**Comment by juherr** [comment]
The nested case seems to be covered by a test: https://github.com/cbeust/testng/pull/2631/files#diff-7ad2353d6285f5aee773e1a24ebe5b4fb75277928ec7d685be88d14932db1083

Could you share a similar TestNG test which highlight your issue?

**Comment by baubakg** [comment]
Hi @juherr ,
Thanks. the tests are actually mine :) I took your code and it ran successfully. However when paying more attention to your implementation I discovered one main difference:

You top level package:
test.annotationtransformer.issue2536.data.*

Mine was:
test.annotationtransformer.issue2536.data

When I changed it locally to include ".*" it works fine. Error in test then.  :) Thanks for taking the time. I'll close the issue.
